tests/hazmat/backends/test_openssl.py

```diff
@@ -478,6 +480,41 @@ def test_unsupported_ec_keys(self):
             backend.create_x509_csr(object(), private_key, hashes.SHA1())
 
 
+class TestOpenSSLSignX509Certificate(object):
+    def test_requires_certificate_builder(self):
+        private_key = RSA_KEY_2048.private_key(backend)
+
+        with pytest.raises(TypeError):
+            backend.sign_x509_certificate(object(), private_key, DummyHash())
+
+    def test_checks_for_unsupported_extensions(self):
+        private_key = RSA_KEY_2048.private_key(backend)
+        builder = x509.CertificateBuilder().subject_name(x509.Name([
+            x509.NameAttribute(x509.OID_COUNTRY_NAME, u'US'),
```

Comment:
We shoudl slim down these tests where possible. There are a lot of cases where a bunch of NameAttributes (or lots of values in an extension) are set that are unnecessary.